projects/saturn/src/main/java/io/fairspace/saturn/webdav/DavFactory.java

```diff
@@ -170,7 +170,24 @@ public boolean isFileSystemResource(org.apache.jena.rdf.model.Resource resource)
         return createResource(type);
     }
 
-    public void createLinkedEntity(String name, org.apache.jena.rdf.model.Resource linkedDirectory, org.apache.jena.rdf.model.Resource type) {
+    public void addLinkedEntity(String name, org.apache.jena.rdf.model.Resource linkedDirectory, org.apache.jena.rdf.model.Resource type) {
+        linkedDirectory.addProperty(FS.linkedEntityType, type);
+
+        var entityIri = linkedEntityIri();
+
+        if(entityIri != null && !entityIri.isEmpty()) {
+            addExistingEntity(linkedDirectory, entityIri);
```

Comment:
it is missing the IRI validation - checking if an entity with this IRI really exist in the database, if it's of type that was added in the linkedEntityType header, if it's not deleted etc...

Reply:
actually I'm not sure if both headers should be send - if linkedEntityIRI header is added, the type seems redundant